src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java

```diff
@@ -205,6 +206,38 @@ protected void positionAndShowDialog(Command onCompleted)
    {
       super.center();
       onCompleted.execute();
+
+      // Force the contents of the modal to be vertically scrollable
+      // if the height of the modal is larger than the app window
+      Element e = this.getElement();
+      Element child = e.getFirstChildElement();
+      if (child != null)
+      {
+         int windowInnerHeight = Integer.parseInt(JsWindow.getProp("innerHeight").toString());
```

Comment:
Instead of adding JsWindow.java and using JSNI, we _could_ do this via 
```
import elemental.client.Browser;

int windowInnerHeight = Browser.getWindow().getInnerHeight();
```
You'd also need to add this to Rstudio.gwt.xml:
```
   <!-- Elemental module dependencies                              -->
   <inherits name='elemental.Elemental'/>
```

Background: `elemental` is a Google project to provide type-safe access to browser APIs from Java with very little overhead. This project has a life beyond GWT (2.x), and is still under active development.

The version bundled with GWT is ancient, but sufficient for basic stuff like this, and if we hit anything it can't do (probably in 1.4), we could switch to the actively maintained version (elemental2: https://github.com/google/elemental2).

The post-GWT 2.x world (in terms of Java-based JavaScript) looks like it will consist of J2CL (https://github.com/google/j2cl), JsInterop (https://github.com/google/j2cl/blob/master/docs/jsinterop-by-example.md), and elemental2, and stuff built on these.

Reply:
I think this is worth a conversation (maybe on #rstudio-dev). I'd definitely be on board adding `elemental` or `elemental2` but want to make sure there isn't some historical reason why we chose not to do so.